### PR TITLE
[GIT] Add test for batched generation

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

Related to #21087, I've added a test for batched generation with GIT.

### CODE REVIEW

File: tests/models/git/test_modeling_git.py [hunk 1]
```diff
@@ -495,3 +495,23 @@ def test_visual_question_answering(self):
...
         self.assertEqual(generated_ids.shape, expected_shape)
         self.assertEquals(generated_caption, "what does the front of the bus say at the top? special")
+
+    def test_batched_generation(self):
+        processor = GitProcessor.from_pretrained("microsoft/git-base-coco")
+        model = GitForCausalLM.from_pretrained("microsoft/git-base-coco")
+        model.to(torch_device)
+
+        image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+        pixel_values = processor(images=image, return_tensors="pt").pixel_values
```
Comment (ydshieh) [inline tests/models/git/test_modeling_git.py:505]: would `images=[image, image]` work for creating a batch of size `2`? Not so important, just to make things shorter. If you decide to change, remember to move `# create batch of size 2` above this line.

File: tests/models/git/test_modeling_git.py [hunk 2]
```diff
@@ -495,3 +495,23 @@ def test_visual_question_answering(self):
...
+
+        # create batch of size 2
+        pixel_values = pixel_values.repeat(2, 1, 1, 1)
+        pixel_values = pixel_values.to(torch_device)
+
+        start_token_id = model.config.bos_token_id
+        generated_ids = model.generate(
+            pixel_values=pixel_values, input_ids=torch.tensor([[start_token_id], [start_token_id]]), max_length=50
```
Comment (ydshieh) [inline tests/models/git/test_modeling_git.py:513]: Probably a good idea to make a comment like 
```python
# we have to prepare `input_ids` with the same batch size as `pixel_values`
```
Let you decide!

**Review by ydshieh — APPROVED** [review]
Nice!

**Comment by NielsRogge** [comment]
Failing test is unrelated, merging.

**Pull request merged**